File: lib/src/main/java/com/sd/lib/dldmgr/processor/impl/TakeFileProcessor.java
package com.sd.lib.dldmgr.processor.impl;

import com.sd.lib.dldmgr.IDownloadDirectory;

import java.io.File;

/**
 * 移动文件处理器
 */
public class TakeFileProcessor extends BaseFileProcessor
{
    public TakeFileProcessor(IDownloadDirectory directory)
    {
        super(directory);
    }

    @Override
    public void process(File file)
    {
        mDirectory.takeFile(file);
    }
}

File: lib/src/main/java/com/sd/lib/dldmgr/processor/impl/TakeFileProcessor.kt
package com.sd.lib.dldmgr.processor.impl

import com.sd.lib.dldmgr.directory.IDownloadDirectory
import java.io.File

/**
 * 移动文件处理器
 */
class TakeFileProcessor(directory: IDownloadDirectory) : BaseFileProcessor(directory) {
    override fun process(file: File) {
        directory.takeFile(file)
    }
}